File: app/src/main/java/com/icaboalo/devfapp/ui/activity/MainActivity.java
package com.icaboalo.devfapp.ui.activity;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.NavigationView;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;

import com.icaboalo.devfapp.R;
import com.icaboalo.devfapp.ui.fragment.ChallengeListFragment;
import com.icaboalo.devfapp.ui.fragment.EventListFragment;
import com.icaboalo.devfapp.ui.fragment.FileListFragment;
import com.icaboalo.devfapp.util.VUtil;

import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnItemClick;

public class MainActivity extends AppCompatActivity {

    @Bind(R.id.app_bar)
    Toolbar mToolbar;

    @Bind(R.id.drawer_layout)
    DrawerLayout mDrawerLayout;

    @Bind(R.id.navigation_view)
    NavigationView mNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        setSupportActionBar(mToolbar);
        replaceFragment(new FileListFragment());
        navigationViewClick();
    }

    private void navigationViewClick(){
        mNavigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(MenuItem item) {
                Fragment nFragment = null;
                switch (item.getItemId()) {
                    case R.id.files:
                        nFragment = new FileListFragment();
                        break;
                    case R.id.events:
                        nFragment = new EventListFragment();
                        break;
                    case R.id.challenges:
                        nFragment = new ChallengeListFragment();
                        break;
                }
                replaceFragment(nFragment);
                mDrawerLayout.closeDrawers();
                return false;
            }
        });
    }

    private void replaceFragment(Fragment fragment) {
        FragmentManager fragmentManager = getSupportFragmentManager();
        fragmentManager.beginTransaction().replace(R.id.container, fragment).commit();
    }
}

File: app/src/main/java/com/icaboalo/devfapp/ui/activity/MainActivity.kt
package com.icaboalo.devfapp.ui.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.view.View
import com.icaboalo.devfapp.R
import com.icaboalo.devfapp.io.model.OptionApiModel
import com.icaboalo.devfapp.ui.adapter.OnViewHolderClick
import com.icaboalo.devfapp.ui.adapter.OptionRecyclerAdapter
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    var optionRecyclerAdapter: OptionRecyclerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupRecycler()
    }

    fun createOptions(): ArrayList<OptionApiModel>{
        val options = ArrayList<OptionApiModel>()
        options.add(OptionApiModel("Search", "", 10))
        options.add(OptionApiModel("Events", "", 10))
        options.add(OptionApiModel("Files", "", 10))
        options.add(OptionApiModel("Skills", "", 10))
        return options
    }

    fun setupRecycler(){
        optionRecyclerAdapter = OptionRecyclerAdapter(this@MainActivity, createOptions(),  object: OnViewHolderClick {
            override fun onClick(v: View, position: Int) {

            }
        })
        val linearLayout = LinearLayoutManager(this@MainActivity)
        recycler_view.adapter = optionRecyclerAdapter
        recycler_view.layoutManager = linearLayout
    }
}
